src/main/kotlin/com/openlattice/organizations/ExternalDatabaseManagementService.kt

```diff
@@ -0,0 +1,369 @@
+        } else {
+            ""
+        }
+        val newColumns = mutableSetOf<OrganizationExternalDatabaseColumn>()
+        assemblerConnectionManager.connect(dbName).use { dataSource ->
+            dataSource.connection.createStatement().use { stmt ->
+                val rs = stmt.executeQuery(
+                        "SELECT information_schema.tables.table_name, information_schema.columns.column_name, " +
+                                "information_schema.columns.data_type, information_schema.columns.ordinal_position, " +
+                                "information_schema.table_constraints.constraint_type " +
+                                "FROM information_schema.tables " +
+                                "LEFT JOIN information_schema.columns on information_schema.tables.table_name = " +
+                                "information_schema.columns.table_name " +
+                                "LEFT OUTER JOIN information_schema.constraint_column_usage on " +
+                                "information_schema.columns.column_name = information_schema.constraint_column_usage.column_name " +
+                                "LEFT OUTER JOIN information_schema.table_constraints " +
+                                "on information_schema.constraint_column_usage.constraint_name = " +
+                                "information_schema.table_constraints.constraint_name " +
+                                "WHERE information_schema.columns.table_name = '$tableName' " +
+                                "$columnCondition"
+                )
+                while (rs.next()) {
+                    val columnName = rs.getString("column_name")
+                    val dataType = rs.getString("data_type")
+                    val position = rs.getInt("ordinal_position")
+                    val isPrimaryKey = rs.getString("constraint_type") == "PRIMARY KEY"
+                    val newColumn = OrganizationExternalDatabaseColumn(
+                            Optional.empty(),
+                            columnName,
+                            columnName,
+                            Optional.empty(),
+                            tableId,
+                            orgId,
+                            dataType,
+                            isPrimaryKey,
+                            position)
+                    newColumns.add(newColumn)
+                }
+            }
+        }
+        return newColumns
+    }
+
+    /*PRIVATE FUNCTIONS*/
+    private fun createPrivilegesSql(action: Action, privileges: List<String>, tableName: String, columnName: String, dbUser: String): String {
+        val privilegesAsString = privileges.joinToString(separator = ", ")
+        val tableNamePath = "$PUBLIC_SCHEMA.$tableName"
+        return if (action == Action.REMOVE) {
+            "REVOKE $privilegesAsString $columnName ON $tableNamePath FROM $dbUser"
+        } else {
```

Comment:
I think there should be a check for `Action.ADD` here and not accept all other actions

Reply:
Earlier in the flow there is a filter that only lets through ADD, SET, and REMOVE I believe. I can still add the check though if it's still better to be safe.

Reply:
Yeah, I would do the explicit check, since the filter is not used in `executePrivilegesUpdate`, and that is a public function.